fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -89,44 +92,45 @@ public FastSerdeCache(Executor executorService, Supplier<String> compileClassPat
   }
 
   public FastSerdeCache(String compileClassPath) {
-    this();
-    this.compileClassPath = Optional.ofNullable(compileClassPath);
+    this(null, compileClassPath);
   }
 
   /**
    *
    * @param executorService
    *            customized {@link Executor} used by serializer/deserializer compile threads
-   * @param compileClassPath
-   *            custom classpath as string
    */
+  public FastSerdeCache(Executor executorService) {
+    this(executorService, (String) null);
```

Comment:
Is this string type cast necessary?

Reply:
Yes - we need to distinguish c-tors:
```
public FastSerdeCache(Executor executorService, Supplier<String> compileClassPathSupplier)
```
and 
```
public FastSerdeCache(Executor executorService, String compileClassPath)
```